libgalois/src/PropertyGraph.cpp

```diff
@@ -116,21 +116,19 @@ WriteTopology(const katana::GraphTopology& topology) {
   }
 
   if (num_nodes) {
-    const auto* raw = topology.out_indices->raw_values();
+    const auto* raw = topology.adj_data();
     static_assert(std::is_same_v<std::decay_t<decltype(*raw)>, uint64_t>);
-    auto buf = std::make_shared<arrow::Buffer>(
-        reinterpret_cast<const uint8_t*>(raw), num_nodes * sizeof(uint64_t));
+    auto buf = arrow::Buffer::Wrap(raw, num_nodes);
     aro_sts = ff->Write(buf);
     if (!aro_sts.ok()) {
       return tsuba::ArrowToTsuba(aro_sts.code());
     }
   }
 
   if (num_edges) {
-    const auto* raw = topology.out_dests->raw_values();
+    const auto* raw = topology.dest_data();
     static_assert(std::is_same_v<std::decay_t<decltype(*raw)>, uint32_t>);
-    auto buf = std::make_shared<arrow::Buffer>(
-        reinterpret_cast<const uint8_t*>(raw), num_edges * sizeof(uint32_t));
+    auto buf = arrow::Buffer::Wrap(raw, num_edges);
```

Comment:
nit: I'd prefer a similar name instead of `ProjectAsArrowArray`. Maybe `WrapAsArrowArray`?